src/evm/protocol/uniswap_v4/state.rs

```diff
@@ -1853,4 +1941,243 @@ mod tests {
         // 500 + 0 - (500 * 0 / 1_000_000) = 500
         assert_eq!(result, 500);
     }
+
+    // Helper to create a basic test pool for swap_to_price tests
+    fn create_basic_v4_test_pool() -> UniswapV4State {
+        use crate::evm::protocol::utils::uniswap::{
+            sqrt_price_math::get_sqrt_price_q96, tick_math::get_tick_at_sqrt_ratio,
+        };
+
+        let liquidity = 100_000_000_000_000_000_000u128; // 100e18
+        let sqrt_price = get_sqrt_price_q96(U256::from(20_000_000u64), U256::from(10_000_000u64))
+            .expect("Failed to calculate sqrt price");
+        let tick = get_tick_at_sqrt_ratio(sqrt_price).expect("Failed to calculate tick");
+
+        let ticks = vec![TickInfo::new(0, 0).unwrap(), TickInfo::new(46080, 0).unwrap()];
+
+        UniswapV4State::new(
+            liquidity,
+            sqrt_price,
+            UniswapV4Fees { zero_for_one: 0, one_for_zero: 0, lp_fee: 3000 }, // 0.3% fee
+            tick,
+            60, // tick spacing
+            ticks,
+        )
+        .expect("Failed to create pool")
+    }
+
```

Comment:
I don't see any test that asserts for an exact amount. Are these tests copied from the v3 tests? 

I think we need some exact assertions in some of the tests, and we also need some test that calls swap_to_price and later get_amount_out and compares the values